fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaViewModel.java

```diff
@@ -152,8 +155,10 @@ public Single<MessageGeneral> sendChirp(
 
   public Single<MessageGeneral> deleteChirp(MessageID chirpId, long timestamp) {
     Log.d(TAG, "Deleting the chirp with id: " + chirpId);
-    Lao lao = getCurrentLao();
-    if (lao == null) {
+    final Lao lao;
+    try {
+      lao = getCurrentLao();
+    } catch (UnknownLaoException e) {
       Log.e(TAG, LAO_FAILURE_MESSAGE);
       return Single.error(new UnknownLaoException());
     }
```

Comment:
Do you think it would be better to move the rest pf the method inside the try block from a readability point of view ?

Reply:
I personally prefer it like this. IMO this snippet is quite explicit in its purpose, i.e. retrieving the LAO and returning an error if absent.

Reply:
Ok, fair enough